miden-lib/asm/sat/layout.masm

```diff
@@ -0,0 +1,396 @@
+export.get_acct_data_ptr
+    push.ACCT_DATA_SECTION_OFFSET
+end
+
+#! Returns the account id.
+#!
+#! Stack: []
+#! Output: [acct_id]
+#!
+#! - acct_id is the account id.
+export.get_acct_id
+    push.ACCT_ID_AND_NONCE_PTR mem_load
+end
+
+#! Returns the account nonce.
+#!
+#! Stack: []
+#! Output: [acct_nonce]
+#!
+#! - acct_nonce is the account nonce.
+export.get_acct_nonce
+    padw push.ACCT_ID_AND_NONCE_PTR mem_loadw
+    movdn.3 drop drop drop
+end
+
+#! Sets the account nonce.
+#!
+#! Stack: [acct_nonce]
+#! Output: []
+#!
+#! - acct_nonce is the account nonce.
+export.set_acct_nonce
+    padw push.ACCT_ID_AND_NONCE_PTR mem_loadw
+    drop movup.3 push.ACCT_ID_AND_NONCE_PTR mem_storew dropw
+end
+
+
+# CREATED NOTES
+# -------------------------------------------------------------------------------------------------
+
+#! Computes a pointer to the memory address at which the data associated with a created note with
+#! index i is stored.
+#!
+#! Stack: [i]
+#! Output: [ptr]
+#!
+#! - i is the index of the created note.
+#! - ptr is the memory address of the data segment for created note i.
+export.get_created_note_ptr
+    exec.constants::get_note_mem_size mul push.CREATED_NOTE_SECTION_OFFSET add
+end
+
+#! Returns the created note recipient
+#!
+#! Stack: [created_note_data_ptr]
+#! Output: [R]
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - R is the recipient of the created note.
+export.get_created_note_recipient
+    padw
+    movup.4 push.CREATED_NOTE_RECIPIENT_OFFSET add
+    mem_loadw
+end
+
+#! Returns the number of assets in the created note
+#!
+#! Stack: [created_note_data_ptr]
+#! Output: [num_assets]
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - num_assets is the number of assets in the created note.
+export.get_created_note_num_assets
+    push.CREATED_NOTE_NUM_ASSETS_OFFSET add mem_load
+end
+
+#! Returns a pointer to the created note asset data
+#!
+#! Stack: [created_note_data_ptr]
+#! Output: [asset_data_ptr]
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - asset_data_ptr is the memory address at which the created note asset data begins.
+export.get_created_note_asset_data_ptr
+    push.CREATED_NOTE_ASSETS_OFFSET add
+end
+
+#! Sets the created note vault hash
+#!
+#! Stack: [created_note_data_ptr, V]
+#! Output: []
+#!
+#! - created_note_data_ptr is the memory address at which the created note data begins.
+#! - V is the vault hash of the created note.
+export.set_created_note_vault_hash
+    push.CREATED_NOTE_VAULT_HASH_OFFSET add mem_storew
+end
+
+# CONSUMED NOTES
+# -------------------------------------------------------------------------------------------------
+
+#! Sets the number of consumed notes.
+#!
+#! Stack: [num_consumed_notes]
+#! Output: []
+#!
+#! - num_consumed_notes is the number of consumed notes.
+export.set_num_consumed_notes
```

Comment:
Should we clarify whether this is the total number of consumed notes or the number of notes consumed so far in the transaction?